### PR TITLE
Chsim dataset

### PR DESCRIPTION
**Pull request by asedlmayr**
Pull request for the chsim dataset file import

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-[![Build Status](https://falcon.ecg.rice.edu:443/buildStatus/icon?job=github_public_agora%2Fdevelop)](https://falcon.ecg.rice.edu:443/job/github_public_agora/job/develop/)
+[![Build Status](https://falcon.ecg.rice.edu:443/buildStatus/icon?job=github_public_agora%2Fchsim-factory)](https://falcon.ecg.rice.edu:443/job/github_public_agora/job/chsim-factory/)
```
Comment (asedlmayr) [inline README.md:1]: Fix this for 'develop' for the pr

File: CMakeLists.txt [hunk 2]
```diff
@@ -604,7 +605,7 @@ foreach(test_name IN LISTS UNIT_TESTS)
     $<TARGET_OBJECTS:agora_sources_lib>
     $<TARGET_OBJECTS:shared_txrx_sources_lib>
     $<TARGET_OBJECTS:common_sources_lib>)
-
```
Comment (asedlmayr) [inline CMakeLists.txt:607]: remove this change

File: files/config/examples/dl-mww-1ue.json [hunk 3]
```diff
@@ -10,13 +10,10 @@
   "tx_gain_b": 81,
   "sample_rate": 7.68e6,
   "ofdm_tx_zero_prefix": 160,
-  /* 192 / 160 */
```
Comment (asedlmayr) [inline files/config/examples/dl-mww-1ue.json:13]: revert this file

File: files/config/examples/dl-sim.json [hunk 4]
```diff
@@ -1,20 +1,19 @@
-{
+{ 
+
```
Comment (asedlmayr) [inline files/config/examples/dl-sim.json:2]: revert this file, and you can make a new one for testing

File: files/config/examples/ul-sim.json [hunk 5]
```diff
@@ -1,4 +1,5 @@
 {
+  "max_frame": 1,
```
Comment (asedlmayr) [inline files/config/examples/ul-sim.json:2]: revert this change

File: files/topology/bs-usrp-addr.json [hunk 6]
```diff
@@ -11,5 +11,4 @@
             "192.168.70.8"
         ]
     }
-}
-
+}
```
Comment (asedlmayr) [inline files/topology/bs-usrp-addr.json:14]: revert

File: simulator/channel.cc [hunk 7]
```diff
@@ -6,78 +6,80 @@
...
+
   float snr_lin = std::pow(10, channel_snr_db_ / 10.0f);
   noise_samp_std_ = std::sqrt(kMeanChannelGain / (snr_lin * 2.0f));
   std::cout << "Noise level to be used is: " << std::fixed << std::setw(5)
             << std::setprecision(2) << noise_samp_std_ << std::endl;
 }
 
-Channel::~Channel() = default;
+ChannelModel* Channel::GetChannelModel() {
+  if (sim_chan_model_ == "AWGN") return new AwgnModel(cfg_);
```
Comment (asedlmayr) [inline simulator/channel.cc:32]: put this in an if / else if / else block with {}

File: simulator/channel.cc [hunk 8]
```diff
@@ -6,78 +6,80 @@
...
-        break;
-    }
+    channel_model->UpdateModel();
   }
-  if (is_downlink) {
-    fmat_h = fmat_src * h_.st();
-  } else {
-    fmat_h = fmat_src * h_;
+
+  switch (channel_model->fading_type) {
```
Comment (asedlmayr) [inline simulator/channel.cc:52]: Why the need for fading type?

File: simulator/channel.cc [hunk 9]
```diff
@@ -6,78 +6,80 @@
...
+    }
+
+    default: {
+      AGORA_LOG_ERROR("Invalid Channel model fading type \n");
+
+      break;
+    }
   }
 
+  //std::printf("Y [%d Rows x %d Cols] \n", (int)fmat_h.n_rows, (int)fmat_h.n_cols);
```
Comment (asedlmayr) [inline simulator/channel.cc:76]: remove

File: simulator/channel.cc [hunk 10]
```diff
@@ -6,78 +6,80 @@
...
+                           channel_model->GetMatrix(is_downlink, h_index);
+        fmat_h.insert_rows(h_index, y_);
+      }
+
+      break;
+    }
+
+    default: {
+      AGORA_LOG_ERROR("Invalid Channel model fading type \n");
+
```
Comment (asedlmayr) [inline simulator/channel.cc:71]: remove extra spaces

File: simulator/channel.h [hunk 11]
```diff
@@ -13,6 +13,10 @@
 #include <numeric>
 
 #include "armadillo"
+#include "channel_models/awgn_model.h"      //Testing
```
Comment (asedlmayr) [inline simulator/channel.h:16]: you can add this file path to the cmake.  channel_models  .  Remove the testing comments

File: test/sim_tests/test_e2e_sim.sh [hunk 12]
```diff
@@ -93,4 +93,4 @@ rm ${output_filepath}/test_agora_output.txt
 rm ${output_filepath}/test_user_output.txt
 rm ${output_filepath}/test_output.txt
 rm ${output_filepath}/test_chsim_output.txt
-exit
```
Comment (asedlmayr) [inline test/sim_tests/test_e2e_sim.sh:96]: revert file

File: src/client/ue_worker.cc [hunk 13]
```diff
@@ -29,6 +29,11 @@ static constexpr bool kPrintDownlinkPilotStats = false;
 static constexpr bool kPrintEqualizedSymbols = false;
 static constexpr bool kDebugTxMemory = false;
 
+/*CHSIM Temp*/
+static constexpr bool kBypassFFTData = false;
+static constexpr bool kBypassFFTPilot = false;
```
Comment (asedlmayr) [inline src/client/ue_worker.cc:34]: lets move all the bypass fft code to it's own branch

File: simulator/channel_models/dataset_model.cc [hunk 14]
```diff
@@ -0,0 +1,153 @@
...
+#include <iostream>
+
+#include "logger.h"
+#include "dataset_model.h"
+
+bool kPrintDatasetOutput = true;
+
+DatasetModel::DatasetModel( const Config* config, std::string dataset_path ) : ChannelModel( config , ChannelModel::kSelective )
+{
+
```
Comment (asedlmayr) [inline simulator/channel_models/dataset_model.cc:11]: remove blank lines

**Review by rdoosty — APPROVED** [review]
